### ISSUE TITLE
Quick access modules disappear when trying to edit them

### ISSUE THREAD

**Issue by esm7** [darktable-org/darktable#16478]
### Describe the bug

When I upgraded to Darktable 4.6.0 I had [this issue](https://github.com/darktable-org/darktable/issues/16293), and as a result, eventually decided to take screenshots of my styles and redo them after deleting `data.db` and starting from scratch.
Now I'm facing a new issue. I rely heavily on the quick access panel, but for some reason I can't get it to work in 4.6.1.
I created a new preset ("my") and added to it the modules I use most often: crop (on-off and aspect), exposure (exposure), rotate and perspective (rotation) and tone equalizer (graph).
**Every time I open the "manage module layouts" dialog, some of the selected modules disappear.**

### Steps to reproduce

1. In Darkroom, in the hamburger menu of the modules, click "manage presets".
2. Create a new preset ("my") of the modules.
3. Remove all modules from the quick access panel.
4. Add crop (on-off, aspect), exposure (exposure), rotate and perspective (rotation), tone equalizer (graph).
5. Close the dialog.
6. The correct modules appear in the quick access.
7. Reopen "manage presets". All the modules disappear except "exposure".

### Logfile | Screenshot | Screencast

![Peek 2024-03-17 14-31](https://github.com/darktable-org/darktable/assets/10722656/000ef5a1-0d68-4d2f-a651-840ca77c83e0)

### Where did you obtain darktable from?

distro packaging

### darktable version

4.6.1

### What OS are you using?

Linux

### What is the version of your OS?

Manjaro

### Are you using OpenCL GPU in darktable?

Yes

**Comment by pehar1** [comment]
Can reproduce with fresh install of 4.7.0+780~ged45e36fcc

**Comment by pehar1** [comment]
Tested the following combinations, again with a completely fresh installation of  `4.7.0+780~ged45e36fcc` :

<pre>
1.
crop 		on-off		disappeared

2.
crop 		on-off		disappeared
crop 		aspect		disappeared

3.
exposure	exposure	ok

4.
exposure	exposure	ok
crop 		on-off		disappeared

5.
exposure	exposure	ok
crop 		on-off		disappeared
crop 		aspect		disappeared

6.
exposure		exposure	ok
rotate and perspective	rotation	disappeared

7.
exposure	exposure	ok
tone equalizer	graph		disappered

8.
exposure	exposure	ok
crop 		on-off		disappeared
crop 		aspect		disappeared
tone equalizer	graph		disappeared

9.
exposure		exposure	ok
color balance rgb	contrast	ok
grain			coarseness	disappered

10.
exposure		exposure			ok
color balance rgb	contrast			ok
color balance rgb	global chroma			ok
color balance rgb	brilliance	highlights	ok
color balance rgb	graph				ok
local contrast		on-off				disappeared

11.
exposure		exposure			ok
local contrast		on-off				disappeared
color balance rgb	contrast			ok

12.
local contrast			disappeared
color balance rgb		ok
exposure			ok
lens correction			disappeared
denoise (profiled)		disappeared
white balance			disappeared
astro denoise			disappeared
base curve			disappeared
bloom				disappeared
blurs				disappeared
canvas enlargement		disappeared
censorize			disappeared
color balance			disappeared
filmic				ok
highpass			disappeared
lens correction			disappeared
hot pixels			disappeared
</pre>

This tests show that there are apparently only a few modules where the issue does **not** occur. Of the modules tested, only `filmic rgb`, `color balance rgb` and `exposure` were **not** affected. The issue seems to depend only on the module, not the specific function selected.

**Comment by pehar1** [comment]
@esm7,  I think I have found the cause of the "misbehavior" (**user error**). The problem lies in step 2 (Create a new preset ("my") of the modules).  To do it correctly you have to

- select an existing preset (e.g. workflow: scene referred)
- **click duplicate**
- enter new preset name (my)
- edit module group (quick access)

**Missing the duplication** leads to the behaviour you described. Stupidly, I fell for it too...

**Comment by esm7** [comment]
I confirm that the problem doesn't happen after duplicating a preset and on a personal level it won't bother me anymore, but is this really the intended behavior? Even if it gets a lower priority due to having an easy workaround, I'm not sure that modules disappearing when opening the dialog makes sense to an unsuspecting user...

**Comment by esm7** [comment]
So what is the "new" button meant for?

**Comment by pehar1** [comment]
Is the "new" button really needed? Could not the "duplicate" button always be used to create a new preset?

**Comment by AlicVB** [comment]
Yep, what you describe is clearly a bug. I'll try to have a look soon

**Comment by AlicVB** [comment]
In fact, the problem happens whatever "button" is used to create a new preset.
items are removed if the corresponding module is not present in one of the other tabs (and the happens more easily with "empty" configuration of course)

PR #16506 should fix this
